<!---->
<launch>

  <param name="robot_description" textfile="$(find mechanum_common)/urdf/mechanum_robot.urdf" />

  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" /> 


  <node pkg="mechanum_control_odom" type="mechanum_control_node" name="mechanum_control_node" >
      <param name="arduino_serial" type="string" value="/dev/ttyUSB0" /> 
      <param name="itsDebugMode" type="bool" value="false" /> 
      <param name="frameBase" type="string" value="base_footprint" /> 
      
  </node>

  <include file="$(find rplidar_ros)/launch/rplidar_with_filters.launch" />


      <node name="hector_mapping" pkg="hector_mapping" type="hector_mapping" output="screen">
   		<remap from="map" to="/mapcurrent" />
    	<param name="map_frame" value="/mapcurrent" />
    	<param name="base_frame" value="/base_footprint" />
    	<param name="odom_frame" value="/odom" />
		<param name="pub_odometry" value="true" />
    	<param name="use_tf_scan_transformation" value="true"/>
		<param name="use_tf_pose_start_estimate" value="false"/>
    	<param name="pub_map_odom_transform" value="false" />
    
		<param name="map_resolution" value="0.025"/>
    	<param name="map_size" value="2048"/>
    	<param name="map_start_x" value="0.5"/>
    	<param name="map_start_y" value="0.5" />
		<param name="map_multi_res_levels" value="2" />
    
    	<param name="update_factor_free" value="0.4" />
    	<param name="update_factor_occupied" value="0.9" />    
    	<param name="map_update_distance_thresh" value="0.4" />
    	<param name="map_update_angle_thresh" value="0.06" />
    	<param name="laser_z_min_value" value = "-1.0" />
    	<param name="laser_z_max_value" value = "1.0" />

    	<param name="advertise_map_service" value="true" />
    
    	<param name="scan_subscriber_queue_size" value="5"/>
    	<param name="scan_topic" value="/scan"/>
    	<param name="pub_map_scanmatch_transform" value="false" />
    	<param name="tf_map_scanmatch_transform_frame_name" value="scanmatcher_frame" />
	</node>
	
	<node name="odomtransformer" pkg="mechanum_common" type="odomtransformer.py" output="screen">
		<param name="odom_pose_input" value="/slam_out_pose" />
		<param name="odom_frame" value="/odom" />
		<param name="tf_frame" value="/base_footprint" />
	</node>
 
  
  <include file="$(find mechanum_common)/launch/mechanum_amcl.launch" >
    <arg name="scan_topic" value="/scan"/>
  </include>
  
  
  
  <include file="$(find my_planner)/launch/my_planner13_test.launch" >
    <arg name="scan_topic" value="/scan"/>
  </include>
</launch>
